doc/user-guide.md

```diff
@@ -160,15 +158,11 @@ func (srv *storageSrv) Write(_ context.Context, req *State, ret func(*WriteRespo
 
 There are some important things to note about implementing the server interfaces:
 
-* Reply messages should be returned using the `ret` function.
-* The handlers are run synchronously, in the order that messages are received.
-  This means that a long-running handler will prevent other messages from being handled.
+* Reply messages must be returned using the `ret` function.
+* The handlers run synchronously, in the order messages are received.
```

Comment:
Should probably add that the ordering only applies to messages between a single node and the server. The order of execution between different nodes is not guaranteed. For example:

> The handlers run in the order that messages are received per node. So all messages from a single node are executed in the order they are received, but messages from two different nodes may not be. Handlers run synchronously, and hence a long-running handler will prevent other messages from being handled.